Chapter12/recipe-01/cxx-example/CMakeLists.txt

```diff
@@ -0,0 +1,80 @@
+# define minimum cmake version
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+
+# project name and supported language
+project(recipe-01 CXX)
+
+# for testing we will need the python interpreter
+find_package(PythonInterp REQUIRED)
+
+# we need ExternalProject functionality
+include(ExternalProject)
+
+# fetch pybind11 sources
+ExternalProject_Add(
+  pybind11_sources
+  PREFIX
+    "${PROJECT_BINARY_DIR}/pybind11"
+  GIT_REPOSITORY
+    https://github.com/pybind/pybind11.git
+  GIT_TAG
+    v2.2
+  CONFIGURE_COMMAND
+    ""
+  BUILD_COMMAND
+    ""
+  INSTALL_COMMAND
+    ""
+  CMAKE_ARGS
+    "-DPYTHON_EXECUTABLE:FILEPATH=${PYTHON_EXECUTABLE}"
+  )
+
+# process our source code as if it was an external project
+ExternalProject_Add(
+  account_project
+  INSTALL_DIR
+    ${PROJECT_BINARY_DIR}/install
+  SOURCE_DIR
+    ${PROJECT_SOURCE_DIR}/account
+  BINARY_DIR
+    ${PROJECT_BINARY_DIR}/account
+  CMAKE_ARGS
+    "-DCMAKE_INSTALL_PREFIX=${PROJECT_BINARY_DIR}/install"
+    "-DPYBIND11_DIR=${PROJECT_BINARY_DIR}/pybind11/src/pybind11_sources/"
+    "-DPYTHON_EXECUTABLE:FILEPATH=${PYTHON_EXECUTABLE}"
```

Comment:
Should this be a `CMAKE_CACHE_ARGS`

Reply:
It then fails. I will keep the CMAKE_ARGS for the moment. I don't know why it fails.